platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -171,137 +230,269 @@ public Schema getSchema(CFMetaData metadata) {
+                                                col.append(((BigDecimal) value).doubleValue());
+                                            break;
+                                        case DOUBLE:
+                                            col.append((Double) value);
+                                            break;
+                                        case FLOAT:
+                                            col.append(((Float) value).doubleValue());
+                                            break;
+                                        case COUNTER:
+                                            col.append(CounterContext.instance().total(byteBuff));
+                                            break;
+                                        case TIME:
+                                            // TODO: Convert Time type to Java Time instead of Java Instant
+                                            long myTime = TimeSerializer.instance.deserialize(byteBuff);
+                                            // Downgrade to millis and append 2000-1-1 as the date
+                                            myTime = myTime / 1000000 + 946706400000L;
+                                            col.append(Instant.ofEpochMilli(myTime));
+                                            break;
+                                        case TIMESTAMP:
+                                            col.append(
+                                                    TimestampSerializer.instance.deserialize(byteBuff).toInstant());
+                                            break;
+                                        case DATE:
+                                            long msTime = SimpleDateSerializer
+                                                    .dayToTimeInMillis(ByteBufferUtil.toInt(byteBuff));
+                                            col.append(Instant.ofEpochMilli(msTime));
+                                            break;
+                                        case DURATION:
+                                            Duration duration = DurationSerializer.instance.deserialize(byteBuff);
+                                            if (duration.getMonths() == 0) {
+                                                int days = duration.getDays();
+                                                long nanos = duration.getNanoseconds();
+                                                long millis = (nanos + Duration.NANOS_PER_HOUR * 24 * days) / 1000000;
+                                                col.append(Converters.toDuration(millis));
+                                            } else {
+                                                // java.time.Duration support day and time (but not month and year)
+                                                throw new RuntimeException(
+                                                        "Failed to convert Duration with months != 0");
```

Comment:
The message should rather say that Cassandra Durations with months are not supported.